cli/command/configure/cmd.go

```diff
@@ -0,0 +1,144 @@
+				log.Err(err).Msg("failed to execute configure command")
+				os.Exit(1)
+			}
+		},
+	}
+	cmd.Flags().StringVarP(&cliUsername, "username", "u", "", "username, available on your saucelabs account")
+	cmd.Flags().StringVarP(&cliAccessKey, "accessKey", "a", "", "accessKey, available on your saucelabs account")
+	return cmd
+}
+
+// explainHowToObtainCredentials explains how to get credentials
+func explainHowToObtainCredentials() {
+	fmt.Println("\nDon't have an account ? Signup here https://saucelabs.com/sign-up !")
+	fmt.Printf("Already have an account ? Get your username and access key here: https://app.saucelabs.com/user-settings\n\n\n")
+}
+
+// interactiveConfiguration expect user to manually type-in its credentials
+func interactiveConfiguration() (*credentials.Credentials, error) {
+	explainHowToObtainCredentials()
+	creds := getDefaultCredentials()
+
+	qs := []*survey.Question{
+		{
+			Name: "username",
+			Prompt: &survey.Input{
+				Message: "SauceLabs username",
+				Default: creds.Username,
+			},
+			Validate: func(val interface{}) error {
+				str, ok := val.(string)
+				if !ok {
+					return errors.New("invalid username. Check it here: https://app.saucelabs.com/user-settings")
+				}
+				str = strings.TrimSpace(str)
+				if len(str) < 1 {
+					return errors.New("you need to type a username. Get yours here: https://app.saucelabs.com/user-settings")
+
+				}
+				return nil
+			},
+		},
+		{
+			Name: "accessKey",
+			Prompt: &survey.Input{
+				Message: "SauceLabs access key",
+				Default: creds.AccessKey,
+			},
+			Validate: func(val interface{}) error {
+				str, ok := val.(string)
+				if !ok {
+					return errors.New("invalid access key. Check it here: https://app.saucelabs.com/user-settings")
+				}
+				str = strings.TrimSpace(str)
+				if len(str) < 1 {
+					return errors.New("you need to type an access key. Get yours here: https://app.saucelabs.com/user-settings")
+
+				}
+				return nil
+			},
+		},
+	}
+
+	if err := survey.Ask(qs, creds); err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("\n\n")
+	return creds, nil
+}
+
+// Run starts the configure command
+func Run(cmd *cobra.Command, cli *command.SauceCtlCli, args []string) error {
+	var creds *credentials.Credentials
+	var err error = nil
+
+	if cliUsername == "" && cliAccessKey == "" {
+		creds, err = interactiveConfiguration()
+	} else {
+		creds = &credentials.Credentials{
+			Username: cliUsername,
+			AccessKey: cliAccessKey,
+		}
+	}
+	if err != nil {
+		return err
+	}
+
+	if !creds.IsValid() {
+		fmt.Println("Credentials provided looks invalid and won't be saved.")
+		return nil
+	}
+	if err := creds.Store(); err != nil {
+		return fmt.Errorf("unable to save credentials")
+	}
+	fmt.Println("You're all set ! ")
+	return nil
+}
+
+// getDefaultCredentials returns first the file credentials, then the one founded in the env.
+func getDefaultCredentials() *credentials.Credentials {
+	fileCreds := credentials.FromFile()
```

Comment:
It seems that there is already a function named `Get` to fetch credentials from file and then env. What about using that one?
And I notice for `getDefaultCredentials`, getting from file has higher priority. But for `credentials.Get()`, getting from env has higher priority. Is there any differences between these two different logics?

Reply:
In case there is already a file containing the credentials, it's making more sense to pre-fill with what is already in the file.
Supporting saving from env => file if there is no file is more to be gentle than asking him to navigate to our website to fetch its credentials.

Does it make sense ?

Reply:
OK. I see. It makes sense :)